src/components/title/show/title-show.js

```diff
@@ -98,13 +98,47 @@ class TitleShow extends Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (!prevProps.model.isLoaded && this.props.model.isLoaded) {
-      const filteredPackages = this.getFilteredPackagesFromParams();
+    const {
+      model,
+      location,
+      history,
+    } = this.props;
 
-      this.setState({ // eslint-disable-line react/no-did-update-set-state
-        filteredPackages,
-        packageFilterApplied: !!filteredPackages.length,
-      });
+    const hasLoadedNewRecords = prevProps.model.isLoading !== model.isLoading;
+
+    if (hasLoadedNewRecords && !isEmpty(model.resources.records)) {
+      const searchParams = qs.parse(location.search, { ignoreQueryPrefix: true });
+      const {
+        filter,
+        filteredPackages: filteredPackagesIds,
+      } = searchParams;
+      const { packageIds } = filter || {};
+
+      if (!filteredPackagesIds && packageIds) {
+        const packagesModels = this.getFilteredPackagesFromParams(packageIds);
+
+        this.setState({
+          filteredPackages: packagesModels,
+          packageFilterApplied: !!packagesModels.length,
+        });
+
+        const search = qs.stringify({
```

Comment:
Can we use `handlePackageFilterChange` method to reduce duplication a little?

Reply:
Great idea, I've changed.